scripts/abins/spowdersemiempiricalcalculator.py

```diff
@@ -353,33 +384,103 @@ def _calculate_s_powder_over_k(self, *, angle: float,
             sdata.add_autoconvolution_spectra()
             sdata = sdata.rebin(self._bins)  # Don't need fine bins any more, so reduce cost of remaining steps
 
-            # Apply appropriate q-dependence to each order, along with 1/(n!) term
-            factorials = factorial(range(1, max_dw_order + 1))[:, np.newaxis]
+            # Compute appropriate q-dependence for each order, along with 1/(n!) term
+            factorials = factorial(range(1, max_dw_order + 1))[order_expansion_slice]
             factorials[:min_order + 1] = 1.  # Remove 1/n! for orders that were explicitly calculated
 
-            q2n_over_n_factorial = q2 ** np.arange(1, max_dw_order + 1)[:, np.newaxis] / factorials
-            sdata *= q2n_over_n_factorial
-
+            q2_order_corrections = q2 ** np.arange(1, max_dw_order + 1)[order_expansion_slice] / factorials
         else:
-            q2n = q2 ** np.arange(1, self._quantum_order_num + 1)[:, np.newaxis]
-            sdata *= q2n
-
-            self._report_progress(f"Finished calculating SData to order {self._quantum_order_num} by analytic powder-averaging.",
-                                  reporter=self.progress_reporter)
+            # (order, q, energy)
             max_dw_order = self._quantum_order_num
+            q2_order_corrections = q2**np.arange(1, max_dw_order + 1)[order_expansion_slice]
 
-        if isotropic_fundamentals or (self._quantum_order_num > 1) or autoconvolution:
+        if broaden:
+            self._report_progress("Applying instrumental broadening to all orders with simple q-dependence")
+            broadening_scheme = abins.parameters.sampling['broadening_scheme']
+            sdata = self._broaden_sdata(sdata, broadening_scheme=broadening_scheme)
+
+        self._report_progress("Applying q^2n / n! q-dependence")
+        sdata *= q2_order_corrections
+        sdata.set_q_bins(self._q_bins)
+
+        if self._isotropic_fundamentals or (self._quantum_order_num > 1) or self._autoconvolution:
             self._report_progress(f"Applying isotropic Debye-Waller factor to orders {min_order} and above.",
                                   reporter=self.progress_reporter)
-            iso_dw = self.calculate_isotropic_dw(q2=q2)
+            iso_dw = self.calculate_isotropic_dw(q2=q2[order_expansion_slice[:-1]])
+
+            if len(iso_dw.shape) == 3:
+                # From q, atom, freq -> atom, q, freq
```

Comment:
We need to get a consistent indexing convention at this point so that the same `sdata.apply_dw()` works for 1-D and 2-D data.

The steps to calculate isotropic_dw use quite a bit of numpy broadcasting. In the 2-D case `q2` is a column vector (creating the 2-D map when combined with a row of energy data) while in 1-D case `q2` is a row vector corresponding to energy.
Perhaps it would make more sense to include this axis check and swap inside `calculate_isotropic_dw`? It might be possible to craft the data input to `_isotropic_dw` more carefully to avoid the axis swap, but I doubt it would reduce complexity.

Reply:
it just seemed odd without the context. A check on the axis would probably be good